Add Monitoring v3 Samples

waprin opened the pull request:
Note: Need tests. I am working on a followup
commit to this and should have it ready by EOD.

You already reviewed this code, this is migration
from git-on-borg.

jonparrott commented on monitoring/list_resources.js:
```diff
@@ -0,0 +1,165 @@
+    // (e.g. Compute Engine, App Engine), the credentials retrieved may require
+    // you to specify the scopes you need explicitly.
+    // Check for this case, and inject the Cloud Storage scope if required.
+    if (authClient.createScopedRequired &&
```
Is this needed anymore? The Python client handles it and I thought I'd yelled about it in Node.js.

jmdobry replied:
I don't think it's done for you:

https://github.com/google/google-auth-library-nodejs#example-usage

waprin replied:
Breaks when I remove it so still seems necessary.

jmdobry commented:
Lots of JSHint errors:

https://travis-ci.org/GoogleCloudPlatform/nodejs-docs-samples/jobs/116475506#L1420

JustinBeckwith commented on monitoring/README.md:
```diff
@@ -0,0 +1,39 @@
+
+# Set up your local dev environment
+
+While Monitoring v3 is in EAP, the necessary client libraries are vendored in, so google-api-node-client is not
```
That's super weird.  We can publish the NPM module right now if that's the issue.

waprin replied:
ugg sorry yes README oversight, Jason already did it.

JustinBeckwith commented on monitoring/README.md:
```diff
@@ -0,0 +1,39 @@
+
+## Contributing changes
+
+See CONTRIBUTING.md
```
I don't think we need the contribution and licensing links here, it's covered at the top level repo

waprin replied:
done

JustinBeckwith commented on monitoring/create_custom_metric.js:
```diff
@@ -0,0 +1,227 @@
+        }
+    }, function (error, customMetric) {
+        if (error) {
+            console.log('Error Creating Custom Metric', error);
```
console.error

waprin replied:
done in several places

waprin commented:
Ty for the extra review Justin/Jon (3 names that start with J in the review btw) I addressed all review comments, jshint passes. I think one of the Travis jobs is about to fail due to a flaky Datastore test. Lmk if you think I can merge anyway, trying to help techwriter get something to point to.

JustinBeckwith commented:
Are we in a particular rush to have this in before there are tests?  If there's a good reason, that's cool, but otherwise I'd prefer to have tests with the PR.

waprin commented:
Stackdriver is supposed to launch sometime soon. They are EST though and I haven't heard fromt hem for a while so maybe it will be tomorrow. I am working on the test code commit right now though should be done soon.

waprin commented:
Ok, Travis is passing, but I think we're not in a rush after all. I'll finish the test commit then re-ping you guys, tahnks.

waprin commented:
Maybe slight rush, tomorrow by 8:30 AM. I think we might just go live without Node though. Working on finishing up tests right now.

jonparrott commented:
Merging for now due to urgency of release. @waprin gets an angry bug for tests.

waprin commented:
Thanks. I'll try to get the testing PR up ASAP (hopefully tonight) just wanted to get ready for launch.
